packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts

```diff
@@ -107,6 +107,13 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
     });
   }
 
+  /**
+   * Add a security group to this load balancer
+   */
+  public addSecurityGroup(securityGroup: ec2.ISecurityGroup) {
+    this.securityGroups.push(securityGroup);
```

Comment:
This will not affect the `Connections` object properly. You need to do an `addSecurityGroup` there as well I think.

In fact, why not take an advance out on https://github.com/aws/aws-cdk/issues/7373 and deduplicate the storage, removing `this.securityGroups` in favor of `this.connections.securityGroups` ?

Reply:
Ah! I actually saw the behavior described in #7472 (while looking at `instance.ts`) when I was searching for other examples of this, and copied the behavior from there. My rationale was similar to your reasoning about mutation here: https://github.com/aws/aws-cdk/issues/7373#issuecomment-615089054

That being said, I suppose this behavior (single SG is mutated, other SG(s) are ignored) is equally wonky and slightly less justifiable, so I'll take your suggestion to deduplicate the storage.

Reply:
Yeah I done fucked this up